src/camera.rs

```diff
@@ -11,6 +11,8 @@ use vecmath::{
 };
 use vecmath::col_mat4_mul as mul;
 use vecmath::consts::Radians;
+use cgmath::{ Quaternion, Rotation, FixedArray, BaseFloat };
+use cgmath::Vector3 as CgVector3; 
```

Comment:
cgmath might be deprecated in favor of nalgebra, and Piston uses vecmath which is designed for use in libraries (type agnostic, easy to reason about). Perhaps we should make a simple quaternion library?